std/uni.d

```diff
@@ -8214,6 +8215,162 @@ unittest //12428
     assert(s == "abcdefghij");
 }
 
+
+// generic toUpper/toLower on whole range, returns range
+private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    static if (!is(Unqual!(ElementEncodingType!Range) : const(dchar)))
```

Comment:
It's settled as far as I'm concerned. Removing autodecoding would cause legacy problems, but for new code byDchar is a far better solution.